docs/proposals/scheduling/20240131-pod-level-numa-policy.md

```diff
@@ -0,0 +1,281 @@
+The value of this label can be `""`, `Restricted`, `SingleNUMANode`, `BestEffort`. The default value of this label is `""`. 
+The meaning of these values is same with the value of policy on node.
+
+#### SingleNUMANodeExclusive
+
+To protect the QoS for the pod with `SingleNUMANode` policy, pods that cross numa should not be placed on the same numa 
+with the pod being scheduled on one numa. Sometimes this may lead to situations where workloads that use multiple NUMAs 
+could never be scheduled. Therefore, we allow some critical workloads to break this restriction.
+
+We will introduce a new property in pod `scheduling.koordinator.sh/resource-spec`: 
+`singleNUMANodeExclusive` to reach the goal, the value of this property can be:
+- `preferred`: a numa with a SingleNUMANode pod will not be scheduled another pod with multi-numa if there is another idle numa.
+- `required`: a numa with a SingleNUMANode pod can not be scheduled another pod with multi-numa.
+
+If `SingleNUMANodeExclusive` not set by user, it will be treated as if 'exclusive' were used to ensure that 'SingleNUMANode' 
+is not affected by other policies.
+
+[//]: # (We will introduce a new argument in scheudler's config: `SingleNUMANodeExclusive`, the value of this property 
+can be:)
+
+[//]: # (- `none`: a numa with a SingleNUMANode pod can be scheduled another pod with multi-numa.)
+
+[//]: # (- `besteffort`: a numa with a SingleNUMANode pod will not be scheduled another pod with multi-numa if there is 
+another idle numa.)
+
+[//]: # (- `exclusive`: a numa with a SingleNUMANode pod can not be scheduled another pod with multi-numa.)
+
+[//]: # (`SingleNUMANodeExclusive` should not be set by users, because some users may set all his/her pods as exclusive 
+so that all pods with restricted policy cannot be scheduled.)
+
+#### Examples
+``` yaml
+metadata:
+  annotations:|-
+      {
+        "numaTopologyPolicy": "SingleNUMANode",
+      }
+spec:
+  containers:
+    resource:
+      requests:
+        cpu: 1
+      limits:
+        cpu: 1
+```
+This pod will be allocated with an exclusive cpu.
+
+``` yaml
+metadata:
+  annotations:|-
+      {
+        "numaTopologyPolicy": "SingleNUMANode",
+      }
+spec:
+  containers:
+    resource:
+      requests:
+        cpu: 1
+      limits:
+        cpu: 2
+```
+This pod will be allocated in shared pool.
+
+``` yaml
+metadata:
+  annotations:|-
+      {
+        "numaTopologyPolicy": "SingleNUMANode",
+      }
+spec:
+  containers:
+    resource:
+      requests:
+        cpu: 1
+        nvidia.com/gpu: 1
+      limits:
+        cpu: 1
+        nvidia.com/gpu: 1
+```
+GPU and cpu will be aligned in one NUMA node for this pod.
+
+``` yaml
+spec:
+  containers:
+    resource:
+      requests:
+        cpu: 1
+        nvidia.com/gpu: 1
+      limits:
+        cpu: 1
+        nvidia.com/gpu: 1
+```
+This pod will be allocated in shared pool, gpu is allocated by
```

Comment:
What does gpu share here mean?

Reply:
Sorry, what I meant was we will not bind align gpu and cpu for this pod in one numa. CPU will use the shared pool and GPU will be allocated by device plugin, which is the same behavior as we schedule a simple gpu pod on a normal node.

Reply:
Have updated the proposal